### ISSUE TITLE
Gauges come and go with latest update

### ISSUE THREAD

**Issue by mcderd-coder** [StephanJoubert/home_assistant_solarman#408]
Hi, I have just updated to 1.5.1 after using your code for many months without issue. Since the upgrade the gauges I have to monitor my solar work sometimes, then they change to messages that say "Entity is non-numeric:" if I just leave it, it seems like when it next refreshes the gauges all work, then a few minutes later I just see the Non-numeric messages.

Also, I have some automations set to inform me of the battery percentage and solar produced throughout the day.

So for example I would get an alert to say solar produced is 5Kwh, then later 10Kwh and so on, now every time I get an alert they are all repeated, so when I get a 15Kwh alert I also get a 10Kwh and a 5Kwh alert, this did not happen before the update.

Screen grabs attached

Many Thanks

![Gauges](https://github.com/StephanJoubert/home_assistant_solarman/assets/121652605/98f9a261-8d8c-495a-bd30-9e653441163f)
![non numeric](https://github.com/StephanJoubert/home_assistant_solarman/assets/121652605/11fc9079-7ad0-4b03-b50e-2328a5af8acb)
![alerts](https://github.com/StephanJoubert/home_assistant_solarman/assets/121652605/9febda8d-4332-48b5-b6ee-34bea9c4c41e)

**Comment by samdunne** [comment]
Same thing happening to me

**Comment by EspeeFunsail** [comment]
Also getting this. All values become "NaN" randomly until it updates. Happening after latest update.

**Comment by ChirpyTurnip** [comment]
This could be related to my issue: [405](https://github.com/StephanJoubert/home_assistant_solarman/issues/405)

It seems in your case the connection is recovered (and hence data comes back) whereas in my case (perhaps because I have two inverters) the one that goes offline does not come back and everything stays as non-data / NaN. 

If this happens again look to see if the integration is actually connected to the inverter or offline.....In my case I can simply check my display:

![Inverter Online](https://github.com/StephanJoubert/home_assistant_solarman/assets/85000166/42d965cc-6134-4a69-a89e-5f6b146faa45)

Mileage may vary.....but I went back to 1.4.3 and everything is solid again. So for now I'm ignoring the option to upgrade...

**Comment by mcderd-coder** [comment]
Yes, the latest version seems a bit flaky. I have also reverted to 1.4.3 and all is working as normal.

**Comment by EspeeFunsail** [comment]
How did you revert back to 1.4.3?

**Comment by mcderd-coder** [comment]
I just installed the day befores backup as I had also updated HA, so I was not sure if the problem was with the solarman code or HA

**Comment by ChirpyTurnip** [comment]
> How did you revert back to 1.4.3?

Just got back to HACS, remove the integration, then re-add it. When putting it back wait for the drop down list to populate and then you can select the version you want (e.g. v.1.4.1, 1.4.3, 1.5, 1.5.1 etc). The one you select is the one that installs....

Just note that it will then prompt you to upgrade to the newest version - ignore that (for a long time until there is a newer version to try).

**Comment by ChirpyTurnip** [comment]
> I just installed the day befores backup as I had also updated HA, so I was not sure if the problem was with the solarman code or HA

It's Solarman - any version of HA with 1.5x doesn't work, any version of HA with 1.4.3 seems to work fine....so that's where my money is....  :-)

**Comment by mcderd-coder** [comment]
Thanks for that info, next time I will uninstall and reinstall via HACS 😉

**Comment by EspeeFunsail** [comment]
> > How did you revert back to 1.4.3?
> 
> Just got back to HACS, remove the integration, then re-add it. When putting it back wait for the drop down list to populate and then you can select the version you want (e.g. v.1.4.1, 1.4.3, 1.5, 1.5.1 etc). The one you select is the one that installs....
> 
> Just note that it will then prompt you to upgrade to the newest version - ignore that (for a long time until there is a newer version to try).

Ok thanks. Does this keep all existing settings the same? Don't want it to break my current sensors from Solarman.

**Comment by ChirpyTurnip** [comment]
It did for me and I have two inverters so double the problems that you would have if it went wrong. The other option, that you may like more,  is to go to HACS > Solarman > Triple Dots in the top right > Re-download. Then select version (1.4.3) and hit download. 

Restart HA and you should be on the old version....

**Comment by ilkka-rautiainen** [comment]
Same for me. The reason seems to be that whenever it doesn't get new values from the inverter the values are undefined, before the last received values were used until new values were received. I tend to get some short offline periods during the day and the previous values were quite good approximation.

**Comment by mcderd-coder** [comment]
> Just updated to latest version and this issue is still there. Reverting back to 1.4.3 again...

What version did you update to? Was it 1.5.1 as that is the broken version, I wouldn't expect this to work until 1.5.2 is released.

**Comment by EspeeFunsail** [comment]
Sorry, you are correct. I saw the update in HACS and thought there was a new update.

**Comment by Testmangh** [comment]
I was also having problem, at night all the gauges would go out and the graph showing the current energy would not mark zero... Reverted back to 1.4.3 não everything works, I lost some entities but its fine...

> It did for me and I have two inverters so double the problems that you would have if it went wrong. The other option, that you may like more, is to go to HACS > Solarman > Triple Dots in the top right > Re-download. Then select version (1.4.3) and hit download.
> 
> Restart HA and you should be on the old version....

![image](https://github.com/StephanJoubert/home_assistant_solarman/assets/25848686/b0a72c7d-f83c-4ed4-a5e3-992203a09412)
